datasets/src/generate.rs

```diff
@@ -66,21 +67,23 @@ fn make_blob(
 /// use linfa_datasets::generate::make_dataset;
 /// let feat_distr = Laplace::new(0.5, 5. ).unwrap();
 /// let target_distr = DiscreteUniform::new(0, 5).unwrap();
-/// make_dataset(5, 5, 2, feat_distr, target_distr);
+/// make_dataset(5, 5, feat_distr, target_distr);
 /// ```
 pub fn make_dataset<X, Y>(
     num_rows: usize,
     num_feats: usize,
-    num_targets: usize,
```

Comment:
You're changing this to single-target data right?

Reply:
yes I am.

Reply:
Wouldn't it be more flexible to support multi-target generation? Single-target is just a special-case for multi-target. A target with dimensions `[n, 1]` can be converted into single-target.

Reply:
I suggest adding a dataset converter method for this in `linfa/src/dataset/impl_dataset.rs`

Reply:
I agree. At the time when I pushed the change I thought linfa algs only supported single_target datasets. What I can do is leave it as a multi-target generation and make the suggested converter method in another PR. 

The question is would you rather I finish this work first or the dataset converter method?

Reply:
Dataset converter, which this work depends on.